### PR TITLE
Correctly report founder's reward amount in getblocktemplate prior to Canopy

### PR DESCRIPTION
**Pull request by nuttycom**
Previously this would return incorrect results in the case that the
miner reward was sent to a shielded address. Post-Canopy, the
foundersreward field is removed; this information should be obtained
from getblocksubsidy instead.

### CODE REVIEW

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -417,6 +417,8 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
             "It returns data needed to construct a block to work on.\n"
             "See https://en.bitcoin.it/wiki/BIP_0022 for full specification.\n"
 
+            "\nTo obtain information about founder's reward or funding stream amounts, use 'getblocksubsidy $(getblocktemplate)[\"height\"]'.\n"
```
Comment (str4d) [inline src/rpc/mining.cpp:420]: This isn't actually valid shell, just something I threw together in Slack 😝 Let's try this:
```suggestion
            "\nTo obtain information about founder's reward or funding stream\n"
            "amounts, use 'getblocksubsidy HEIGHT' passing in the height returned\n"
            "by this API.\n"
```

**Review by daira — APPROVED** [review]
utACK

**Review by str4d — APPROVED** [review]
utACK

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**